t/README.md

```diff
@@ -0,0 +1,75 @@
+# Unit Tests for PG
+
+The individual unit tests are located in each of the directories.
+
+Formal unit tests are located in the the `macros` and `contexts` directories that are designed to test the pg macros and contexts respectively.
+
+## Running the tests
+
+```
+cd $PG_ROOT
+prove -r .
+```
+
+will run all of the tests in `.t` files within subdirectories of `t`.
+
+### Running an individual test
+
+If instead, you want to run an individual test, for example the `pgaux.t` test suite,
+```
+cd $PG_ROOT/t/macros
+prove -v pgaux.t
+```
+
+which will be verbose (`-v`).
+
+## Writing a Unit Test
+
+To write a unit test, the following is needed at the top of the file:
+
+```perl
+use warnings;
+use strict;
+package main;
+
+use Test::More;
+use Test::Exception; # needed only if you test for errors.
+
+## the following needs to include at the top of any testing  down to TOP_MATERIAL
+
+BEGIN {
+	die "PG_ROOT not found in environment.\n" unless $ENV{PG_ROOT};
+	die "WEBWORK_ROOT not found in environment.\n" unless $ENV{WEBWORK_ROOT};
+
+	$main::pg_dir = $ENV{PG_ROOT};
+	$main::webwork_dir = $ENV{WEBWORK_ROOT};
+
+}
```

Comment:
Markdown files don't like the hard tab usage here.  They should probably be changed to four spaces.